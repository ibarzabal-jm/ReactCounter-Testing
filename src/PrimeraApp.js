import React from "react";
import PropTypes from 'prop-types';


const PrimeraApp = ( { saludo, subtitulo } ) => {

   
    return(
        <>
            <h1> { saludo } !!!</h1>
           
            <h3>{subtitulo}</h3>
            <p>Mi primera aplicaci√≥n</p>
        </> 
    );

};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo',
}



export default PrimeraApp;
