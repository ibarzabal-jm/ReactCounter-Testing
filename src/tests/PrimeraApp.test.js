import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp'


describe('Pruebas en <PrimeraApp />', () => {
    
    /* test('debe de mostrar el mensaje "Hola, soy Goku"', () => {
        
        const saludo = 'Hola, soy Goku';
        
        const {getByText} = render( <PrimeraApp saludo={saludo} /> )

        expect( getByText( saludo )).toBeInTheDocument();

    }) */
    
    test('debe de mostrar el componente <PrimeraApp/> correctamente', () => {
        
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={saludo} /> )

        expect( wrapper).toMatchSnapshot();

    })
    
    test('debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Soy un subtitulo';
        
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo}
            /> 
        );

        const textoParrafo = wrapper.find('h3').text();
        
        expect( textoParrafo ).toBe(subtitulo);
        
    })
    

})


