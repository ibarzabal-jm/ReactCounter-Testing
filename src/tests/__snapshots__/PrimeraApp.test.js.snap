// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar el componente <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     !!!
  </h1>
  <h3>
    Soy un subtitulo
  </h3>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
